src/Scope.zig

```diff
@@ -248,6 +309,67 @@ pub fn findBlockReturnType(inner: *Scope) aro.QualType {
     }
 }
 
+pub fn processContainerMemberFnsList(t: *Translator, list: std.ArrayListUnmanaged(ContainerMemberFns)) !void {
+    for (list.items) |item| {
+        const decl_tag = item.container_decl.tag();
+        const decl_data = switch (decl_tag) {
+            inline .var_simple, .pub_var_simple => |tag| item.container_decl.castTag(tag).?.data,
+            else => unreachable,
+        };
+
+        var container_data = switch (decl_data.init.tag()) {
+            inline .@"struct", .@"union" => |tag| &decl_data.init.castTag(tag).?.data,
+            else => unreachable,
+        };
+
+        const arena = t.arena;
+        const old_variables = container_data.variables;
+        var new_variables = try arena.alloc(ast.Node, old_variables.len + item.member_fns.items.len);
+        @memcpy(new_variables[0..old_variables.len], old_variables);
+        // Assume the allocator of container_data.variables is arena,
+        // so don't add arena.free(old_variables).
+        var func_ref_vars = new_variables[old_variables.len..];
+        var last_name_array = try t.gpa.alloc([]const u8, item.member_fns.items.len);
```

Comment:
This should be a `StringHashMap(u32)` and it should be outside the main loop to avoid repeated frees and allocations.